assets/js/components/Dashboard/PaMessageForm/SelectStationsAndZones/SelectZonesPage.tsx

```diff
@@ -66,10 +68,18 @@ const SelectZonesPage = ({
         return routeID;
       }),
     )(places);
-  }, [places, routeToRouteIDMap, value]);
+  };
+
+  // eslint-disable-next-line react/hook-use-state
+  const [initialPlacesWithSelectedSigns] = useState<Place[]>(
+    getInitialPlacesWithSelectedSigns,
+  );
```

Comment:
oops, this was [my suggestion](https://github.com/mbta/screenplay/pull/456#discussion_r1731347011) in the first place (omitting the setter assignment entirely to reduce clutter), and I didn't realize a lint rule would complain about it. [Relevant discussion here](https://github.com/jsx-eslint/eslint-plugin-react/issues/3268) — in particular it seems like this rule doesn't have a special case for underscores in the name, nor is it configurable in a way that would allow them. The name just has to be exactly the same but with `set` prepended.

On a second reading I'm not sure why this can't be simplified to:

```ts
const initialPlacesWithSelectedSigns = getInitialPlacesWithSelectedSigns()
```

Reply:
And to Cora's point, it seems like I need to either disable the rule or convert it to a `useMemo` (which seems like it would have the same problem as before due to reevaluation on rerenders).

Reply:
Does this work? It's a slightly weird way to use a ref, but like state, the initial value is only ever assigned once.

```ts
const { current: initialPlacesWithSelectedSigns } = useRef(
  places.filter((p) => p.screens.some((s) => value.includes(s.id))),
);
```

Reply:
Can I suggest that we instead remove `react/hook-use-state` from our eslint config?

https://github.com/mbta/screenplay/blob/6441ed4b513cb7d73e333437ab03fa0c148bfcce/assets/.eslintrc.js#L44

It isn't part of the recommended settings for react hooks which we do have enabled

https://github.com/mbta/screenplay/blob/6441ed4b513cb7d73e333437ab03fa0c148bfcce/assets/.eslintrc.js#L10

and I think through this discussion we've demonstrated a thorough understanding of what we are doing and why.

---

As an aside, I'm being picky about this because "ignore" comments for linting rules feel like a code smell to me. I feel like we should be all in any linting rule we adopt. Anything that requires more nuance than the linter can provide should be handled in code review. And I really like @digitalcora's suggestion to just destructure out the value and only suggested the `_` prefix because I thought eslint was complaining about the unused variable.

Reply:
The `useRef` is interesting. I think I'll go with that. Won't leave us confused why we threw away the setter in `useState`.

And I'm all for removing this rule. I foresee it doing more harm than good.

Reply:
If we're removing the rule I'd go with the `useState`. If anything the `useRef` approach is more confusing, the only advantage being that it keeps to the letter of the rule if we wanted to retain it. (I'd certainly like it if this rule could check setter names only _if_ you destructure them, and allow a setter to be entirely absent, but it doesn't seem like there's a way to set that up.)